eZ/Publish/Core/FieldType/DateAndTime/SearchField.php

```diff
@@ -28,10 +29,17 @@ class SearchField implements Indexable
      */
     public function getIndexData(Field $field, FieldDefinition $fieldDefinition)
     {
+        if (isset($field->value->data['timestamp']) && $field->value->data['timestamp'] !== '') {
```

Comment:
> And we are sure this only comes from storage and not user (via input validation and such)?

Yes. `getIndexData` is called by `eZ\Publish\SPI\Search\Handler::indexContent` which always gets its content from `eZ\Publish\SPI\Persistence\Content\Handler::load`